include/swift/Basic/FileTypes.def

```diff
@@ -98,6 +98,7 @@ TYPE("pch",                 PCH,                       "pch",             "")
 TYPE("none",                Nothing,                   "",                "")
 
 TYPE("abi-baseline-json",   SwiftABIDescriptor,        "abi.json",        "")
+TYPE("api-json",            SwiftAPIDescriptor,        "",                "")
```

Comment:
The extension right now for partial SDKDB files is `.sdkdb` . Should this be listed in the extension field?

Reply:
I could not identify a reason it was important to specify an extension here. There are some utilities in the compiler to take a filename and look up its corresponding `file_types` entry using the contents of this table, but AFAIK, we don't need to do that with an API descriptor since the path to one is always specified explicitly in the arguments to the frontend. We could certainly add `.sdkdb` here but it feels slightly over-specified to me since ultimately this output is just JSON to the compiler, it doesn't otherwise know about SDKDB as a concept.

@artemcm @xymus @nkcsgexi Am I missing any reasons this needs to be specified?

Reply:
Yeah, I don't remember any specific reasons for being explicit about file extensions other than inferring output kinds. However, it seems to be good documentation, no?

Reply:
I think I named it originally `partial.sdkdb` for the intermediate JSON format. Up for a better name.

Reply:
I think given what I've heard that we do not need to encode the suffix of this file in the Swift compiler. From the compiler's point of view it's just another type of supplementary file that it knows how to write to a given path when requested, but the filename can be opaque to the compiler.